delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,409 @@
+                handle_split(alert.get(SSHHandler.OID_LAST_TIME), '=', 1)
+            time_type = '%a %b %d %H:%M:%S %Y'
+            occur_time = int(time.mktime(time.strptime(
+                timestamp,
+                time_type)))
+            alert_model['occur_time'] = int(occur_time * 1000)
+            alert_model['description'] = alert_name
+            alert_model['resource_type'] = SSHHandler.handle_split(
+                alert.get(SSHHandler.OID_OBJ_TYPE), '=', 1)
+            alert_model['location'] = SSHHandler.handle_split(alert.get(
+                SSHHandler.OID_SYS_NAME), '=', 1)
+            return alert_model
+        except Exception as e:
+            LOG.error(e)
+            msg = ("Failed to build alert model as some "
+                   "attributes missing in alert message.")
+            raise exception.InvalidResults(msg)
+
+    def login(self, context):
+        """Test SSH connection """
+        try:
+            with self.ssh_pool.item() as ssh:
+                self.do_exec(SSHHandler.login_command, ssh)
+        except Exception as e:
+            LOG.error("Failed to login ibm storwize_svc %s" %
+                      (six.text_type(e)))
+            raise e
+
+    def do_exec(self, command_str, ssh):
+        """Execute command"""
+        try:
+            utils.check_ssh_injection(command_str)
+            if command_str is not None and ssh is not None:
+                stdin, stdout, stderr = ssh.exec_command(command_str)
+                res, err = stdout.read(), stderr.read()
+                re = res if res else err
+                result = re.decode()
+        except paramiko.AuthenticationException as ae:
+            LOG.error('doexec Authentication error:{}'.format(ae))
+            raise exception.InvalidUsernameOrPassword()
+        except Exception as e:
+            LOG.error('doexec InvalidUsernameOrPassword error:{}'.format(e))
+            if 'timed out' in str(e):
+                raise exception.SSHConnectTimeout()
+            elif 'No authentication methods available' in str(e) \
+                    or 'Authentication failed' in str(e):
+                raise exception.SSHInvalidUsernameOrPassword()
+            elif 'not a valid RSA private key file' in str(e):
+                raise exception.InvalidPrivateKey()
+            elif 'not found in known_hosts' in str(e):
+                raise exception.SSHNotFoundKnownHosts(self.ssh_pool.ssh_host)
+            else:
+                raise exception.SSHException()
+        return result
+
+    def handle_detail_ssh(self, context, command, method):
+        try:
+            with self.ssh_pool.item() as ssh:
+                ssh_info = self.do_exec(command, ssh)
+            return ssh_info
+        except Exception as e:
+            msg = "Failed to ssh ibm storwize_svc %s: %s" % \
+                  (method, six.text_type(e))
+            raise exception.SSHException(msg)
+
+    def handle_capacity(self, value):
```

Comment:
What is `handle_capacity` for? Please change the function name to a meaningful one.

Reply:
changed